app/controllers/scans_controller.rb

```diff
@@ -33,17 +33,30 @@ def current_request
 
   protected
 
+  def illiad_page
+    @illiad_page = { 'Action': '10', 'Form': '30', 'rft.genre': 'scananddeliverArticle' }
+  end
+
+  def illiad_item_data(scan)
+    @illiad_item_data = { 'rft.location': scan.origin,
+                          'rft.callnum': scan.holdings.first.callnumber,
+                          'rft.item': scan.holdings.first.barcode }
+  end
+
+  def illiad_citation(scan)
+    @illiad_params = { 'rft.jtitle': scan.item_title,
+                       'rft.au': scan.data[:authors],
+                       'rft.pages': scan.data[:page_range],
+                       'rft.atitle': scan.data[:section_title] }
+  end
+
+  def illiad_callback(scan)
+    @illiad_callback = { 'scan_referrer': successful_scan_url(scan) }
+  end
+
   def illiad_query(scan)
```

Comment:
This seems to be getting a bit larger than the context of the controller.  It's possible that it may be time to refactor all these methods out of the controller class and into its own class or module.